### PR TITLE
Feature/basic structure

### PR DESCRIPTION
**Pull request by RazorMeister**
I've created basic structure for all needed files.

Added:

- React router
- Redux (it will only be used for authentication)
- Basic structure for pages / components and layouts
- Login posibility
- Simple flats list
- Mockup server (json-server)

Structure description:

- common - directory for every useful helpers, constants, utils, enums, types
- components - here goes simple components
- layouts - layout of the page. For now there are two layouts - for main app and auth pages
- mocks - place for mockup server with sample data
- modules - custom hooks for getting data from api etc.
- pages - pages components. Every page which is connected to react-router should be here
- routes - routes and route-guards
- services - services for api communication
- store - place for redux files
- styles - scss files

Basic concept of creating new page resource with some api calls:

1. Create page (pages directory). If there are be more connected pages add subdirectory like Flats.
2. Add pages to router in routes/CreateRoutes file
3. Create custom hook (for example useFlats) and service (for example FlatService). Pay attention that hooks have plural name and services singular of resource.

Communication scheme: 

**USER** <-> Page (_<-> Component)_ <-> CustomHook <-> Service <-> **BACKEND API**

Commands:

- npm start - start react app
- npm run mock - mock fake backend api

### CODE REVIEW

**Review by lavar3l — APPROVED** [review]

File: src/modules/useAuth.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        });
+        success({ title: "Logged out successfully!" });
+        navigate('/auth/login');
+    }
+
+    return {
+        user,
+        isLoggedIn,
+        login,
+        logout
```
Comment (Brotholomew) [inline src/modules/useAuth.ts:45]: This is not a bug, just tagging code that I don't understand

File: src/modules/useFlats.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                .finally(() => setFlatsLoading(false));
+        })
+    }
+
+    return {
+        flatsLoading,
+        flats,
+        flat,
+        fetchFlats,
+        fetchFlat
```
Comment (Brotholomew) [inline src/modules/useFlats.ts:41]: This is not a bug, just tagging code that I don't understand

File: src/modules/useNotification.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    toast: true,
+    position: 'bottom-end',
+    showConfirmButton: false,
+    timer: 3000,
+    timerProgressBar: true,
+}
+
+const useNotification = () => {
+    const toast = (data: any) => {
+        MySwal.fire({
```
Comment (Brotholomew) [inline src/modules/useNotification.ts:21]: This is not a bug, just tagging code that I don't understand

File: src/services/AxiosInstance.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+AxiosInstance.interceptors.response.use(
+    ({ data }: any) => data,
+    (error: any) => {
+      //(new ErrorService()).handle(error.response?.status, error.response?.data)
+      throw error
+    }
+)
+
+export default AxiosInstance;
```
Comment (Brotholomew) [inline src/services/AxiosInstance.ts:35]: This is not a bug, just tagging code that I don't understand

File: src/styles/utils/button.scss [hunk 5]
```diff
@@ -0,0 +1,10 @@
+.default-button {
+
+  &--info {
+    background-color: #61dafb;
+  }
+
+  &--error {
+    background-color: red;
+  }
+}
```
Comment (Brotholomew) [inline src/styles/utils/button.scss:10]: why are we using scss?

**Review by Brotholomew — APPROVED** [review]

**Pull request merged**